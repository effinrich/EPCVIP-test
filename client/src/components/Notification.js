import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { FormClose, StatusGood } from 'grommet-icons'
import { Box, Button, Layer, Text } from 'grommet'

const NotificationLayer = ({ isReady }) => {
  const [open, setOpen] = useState()

  const onOpen = () => {
    setOpen(true)
    setTimeout(() => {
      setOpen(undefined)
    }, 5000)
  }

  const onClose = () => setOpen(undefined)

  useEffect(() => {
    if (isReady) {
      onOpen()
    }
  }, [isReady])

  return (
    <Layer
      position="top"
      modal={false}
      margin={{ vertical: 'medium', horizontal: 'small' }}
      onEsc={onClose}
      responsive={false}
      plain
    >
      {open && (
        <Box
          align="center"
          direction="row"
          gap="small"
          justify="between"
          round="small"
          elevation="small"
          pad={{ vertical: 'xsmall', horizontal: 'small' }}
          background="status-ok"
        >
          <Box align="center" direction="row" gap="xsmall">
            <StatusGood color="white" />
            <Text color="white">
              A user has been successfully created or modified (this
              notification will close after 5 seconds)
            </Text>
          </Box>
          <Button icon={<FormClose color="white" />} onClick={onClose} plain />
        </Box>
      )}
    </Layer>
  )
}

NotificationLayer.propTypes = {
  isReady: PropTypes.bool
}

export default NotificationLayer
