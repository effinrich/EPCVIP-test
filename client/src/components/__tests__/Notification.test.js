import React from 'react'

import { render } from 'utils/testUtils/helperRtl'

import Notification from '../Notification'

describe('Notification Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<Notification />)
    expect(container).toMatchSnapshot()
  })

  it('should render child element', () => {
    const { container } = render(<Notification />)
    expect(container.firstChild).toBeDefined()
  })
})
