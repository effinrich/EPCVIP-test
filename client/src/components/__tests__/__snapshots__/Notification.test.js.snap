// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component should render without crashing 1`] = `<div />`;
